cmd/krel/cmd/release_notes.go

```diff
@@ -339,65 +370,141 @@ func createDraftPR(tag string) (err error) {
+		// Workdir is where we store the session state and all the release-notes maps
+		releaseDir := fmt.Sprintf(
+			"%s/releases/release-%d.%d", sigReleaseRepo.Dir(),
+			tagVersion.Major, tagVersion.Minor,
+		)
+
+		// Check if the release exists
+		if !util.Exists(releaseDir) {
+			return errors.New(fmt.Sprintf(
+				"could not find release directory for v%d.%d",
+				tagVersion.Major, tagVersion.Minor,
+			))
+		}
+
+		workdir := filepath.Join(releaseDir, mapsDirectory)
+		if !util.Exists(workdir) {
+			logrus.Debugf("Working directory does not exist. Creating.")
+			if err := os.Mkdir(workdir, os.FileMode(0o755)); err != nil {
+				return errors.Wrap(err, "creating release notes workind directory")
+			}
+		}
+
+		// Run the release notes fix flow
+		err := fixReleaseNotes(workdir, releaseNotes)
+		if err != nil {
+			return errors.Wrap(err, "while running release notes fix flow")
+		}
+
+		rnMapProvider, err := notes.NewProviderFromInitString(workdir)
+		if err != nil {
+			return errors.Wrap(err, "creating release notes draft")
+		}
+		for _, note := range releaseNotes.ByPR() {
+			maps, err := rnMapProvider.GetMapsForPR(note.PrNumber)
+			if err != nil {
+				return errors.Wrapf(err, "while getting maps for PR#%d", note.PrNumber)
+			}
+			for _, noteMap := range maps {
+				if err := note.ApplyMap(noteMap); err != nil {
+					return errors.Wrapf(err, "applying note maps to pr #%d", note.PrNumber)
+				}
+			}
+		}
+	}
+
+	// Generate the results struct
+	result, err := buildNotesResult(start, releaseNotes)
+	if err != nil {
+		return errors.Wrap(err, "building release notes results")
+	}
 
 	// generate the notes
-	targetdir := filepath.Join(sigReleaseRepo.Dir(), "releases", fmt.Sprintf("release-%d.%d", s.Major, s.Minor))
+	targetdir := filepath.Join(sigReleaseRepo.Dir(), "releases", fmt.Sprintf("release-%d.%d", tagVersion.Major, tagVersion.Minor))
 	logrus.Debugf("Release notes markdown will be written to %v", targetdir)
 	err = ioutil.WriteFile(filepath.Join(targetdir, draftFilename), []byte(result.markdown), 0644)
 	if err != nil {
 		return errors.Wrapf(err, "writing release notes draft")
 	}
+	logrus.Infof("Release Notes Draft written to %s", filepath.Join(targetdir, draftFilename))
+
+	// If we are in interactive mode, ask before continuing
+	if !autoCreatePullRequest {
+		autoCreatePullRequest, err = util.AskYorN("Create pull request with your changes?")
+		if err != nil {
+			return errors.Wrap(err, "while asking to create pull request")
+		}
+	}
+
+	if !autoCreatePullRequest {
+		fmt.Println("Pull request creation aborted. Your local copy of k/sig-release")
+		fmt.Println("has note been deleted for you to review your changes. You may")
+		fmt.Println("push your changes to your fork and create the PR from there.")
+		fmt.Println("\nYour fork of kubernetes/sig-release was cloned here:")
+		fmt.Println(sigReleaseRepo.Dir())
+		logrus.Warn("Pull request aborted. Local changes were not pushed back.")
+		return nil
```

Comment:
- do we favor this format over the multiline string?
- typo: "has **note** been deleted" --> "has **not** been deleted"
    - I think I would find it helpful with elaborate _what to do next_, something like:
      ```
      Pull request has NOT been created. The changes were made to your local copy of k/sig-release.
      To complete the process, you will need to:
        1. Review the changes in your local copy
        2. Push the changes to your fork (git push -u userfork <branch>)
        3. Submit a pull request to k/sig-release

      You can find your local copy here:
      /path/to/repository

      [WARN] Changes were made locally, user need to perform manual push and create pull request.
      ```
      what do you think? I'm raising this for discussion, not a grading rubric 😄

Reply:
Thanks @wilsonehusin this is precisely the suggestions I need: straight from the battle lines :) 
I like it better like you're suggesting, I'll add this in the next commit.

Reply:
Thanks for raising the point @JamesLaverack , I would have never known by myself that "aborted" should be avoided. Also glad to see wg-naming starting to have influence over what we're doing :)

Done, I've incorporated @wilsonehusin suggestion